### ISSUE TITLE
A feature request：How can I judge user intentions ？

### ISSUE THREAD

**Issue by gangyahaidao** [kaldi-asr/kaldi#2614]
Hello，
I have a need for speech recognition now, and I have read many documents of this project, but I am still not sure whether this project can meet my need：

Now I have hundreds of thousands of wav audio files, which are only one to five seconds and divided into two categories, one is positive answer, the other is negative answer, but I do not have the text information corresponding to each wav file, now my demand is whether I can use this project to make intention judgment？

For example, if I input an audio data, then I can get the intention expressed by this audio, but there is no text corresponding to this audio

Any help will be greatly appreciated！

**Comment by danpovey** [comment]
It might be easier to use a more generic machine learning toolkit, since Kaldi is more oriented to tasks where there are some kind of sequences involved.  E.g. you could extract features for fixed-length chunks of those files and train a CNN (i.e. treat them as images, num-filter-banks by num-frames, to be classified).  Then for test utterances you could just average over all chunks.

If you don't know what filter-banks or frames are then maybe read the first bits of the HTK Book.  Kaldi is usable to extract features, but you need to at least understand the basic terminology first.

**Comment by gangyahaidao** [comment]
Thank you very much for your reply, I understand the HTK speech recognition toolbox, may be able to meet my needs, but the final recognition accuracy need to be further test, anyway, I will try these two ways, one is direct to the audio file parsing recognize intention, another is the use of audio and text in-depth study

**Comment by danpovey** [comment]
I wasn't talking about using HTK, I was talking about things like TensorFlow.  I was talking about the HTK Book only for background information so you know speech-related terminology.

**Comment by gangyahaidao** [comment]
Got it
I have a training plan to mark ten thousand data, using the mature training tools, such as Kaldi or DeepSpeech, theoretically, make corresponding processing to the audio data, can use neural network training voice and data as training images, either way need to try, after all this demand, so far I have not found a may follow the example

**Comment by gangyahaidao** [comment]
Hi, @danpovey 
You are a master. I don't know whether my demand is feasible in theory, that is, I use deep learning to deal with audio features without accurate audio annotation of text, and then I use this model to judge the user's speech intention. Is that possible

**Comment by danpovey** [comment]
It's certainly possible but right now you don't seem to have much experience.  You won't be able to do it unless you get familiar with the tools.  I don't have time to answer generic questions like this.

**Comment by gangyahaidao** [comment]
I will keep leaning and try it, Thank you for your advice
